src/inmanta/execute/proxy.py

```diff
@@ -77,27 +98,44 @@ class ProxyContext:
     :param allow_reference_values: Allow references for values accessed through this proxy. Either because they have been
         declared and validated, or because explicitly requested. Defaults to allow references iff the object has been
         validated. Kept as a separate field to allow overrides for a single proxy object while maintaining behavior for
-        nested proxies.
+        nested proxies. This value is not passed on to children.
     :param path: The path, within the plugin's argument namespace, where this object lives. Should be compatible with
         composition at the tail, e.g. `list_arg[0].attr`.
     """
 
-    path: str
-    mode: ProxyMode = ProxyMode.PLUGIN
-    validated: bool = True
-    allow_reference_values: Optional[bool] = None
+    __slots__ = ("path", "validated", "allow_reference_values")
 
-    def nested(self: Self, *, relative_path: str) -> Self:
+    def __init__(self, path: str, validated: bool = True, allow_reference_values: Optional[bool] = None) -> None:
```

Comment:
I quite liked the `kw_only` aspect of this one tbh

```suggestion
    def __init__(self, *, path: str, validated: bool = True, allow_reference_values: Optional[bool] = None) -> None:
```